common/plugins/com.liferay.ide.core/src/com/liferay/ide/core/util/LaunchHelper.java

```diff
@@ -357,7 +359,7 @@ protected void addUserEntries( RuntimeClasspathModel model ) throws CoreExceptio
             }
             else if( userEntry instanceof IClasspathEntry )
             {
-                entry = new RuntimeClasspathEntry( userEntry );
+                entry = new LiferayRuntimeClasspathEntry( userEntry );
```

Comment:
Should create the old classpathEntry in the old way and pass that object to a new wrapper class.